IdentityCore/src/cache/metadata_cache/MSIDMetadataCacheAside.m

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDMetadataCacheAside.h"
+#import "MSIDMetadataCacheDataSource.h"
+#import "MSIDCache.h"
+#import "MSIDJsonSerializable.h"
+#import "MSIDJsonSerializer.h"
+#import "MSIDJsonSerializing.h"
+#import "MSIDCacheKey.h"
+
+@implementation MSIDMetadataCacheAside
+{
+    NSMutableDictionary *_memoryCache;
+    id<MSIDMetadataCacheDataSource> _dataSource;
+    dispatch_queue_t _synchronizationQueue;
+    MSIDJsonSerializer *_jsonSerializer;
+}
+
+- (instancetype)initWithDataSource:(id<MSIDMetadataCacheDataSource>)dataSource
+{
+    if (!dataSource) return nil;
```

Comment:
nit: create error?

Reply:
not sure if this is needed here,
1. this shouldn't be a blocker for anyone even if this is not supplied, 
2. seems like other accessors we create do not create error, we can create an enhancement tag to have all our accessors create error if it fails to create one.